modules/auxiliary/scanner/scada/modbusclient.rb

```diff
@@ -369,6 +379,31 @@ def write_registers
       print_error("Unknown answer")
     end
   end
+  
+  def read_id
+    @function_code = 0x2b
+    print_status("Sending READ ID...")
+    response = send_frame(make_read_id_payload)
+    if response.nil?
+      print_error("No answer for the READ REGISTER")
+    elsif response.unpack("C*")[7] == (0x80 | @function_code)
+      handle_error(response)
+    elsif response.unpack("C*")[7] == @function_code
+      @ObjCnt = response[13].unpack("C")[0]
+      @ObjIdPos=14
+      begin
+	@ObjLen = response[@ObjIdPos+1].unpack("C")[0]
+	value = response.slice(@ObjIdPos+2,@ObjLen).unpack("a*")[0]
+ 	objid=response[@ObjIdPos].unpack("C")[0]
+	print_good("Object ID #{objid}: #{value}")
+	@ObjCnt = @ObjCnt-1
+	@ObjIdPos = @ObjIdPos + @ObjLen + 2
+     end while @ObjCnt > 0
+    else
+      print_error("Unknown answer")
```

Comment:
Maybe print out the value you get; basically dump the value as a hex string to the screen.

Reply:
I guess we could do that of course. Actually I didn't want to change the whole module as every existing action in this modules behaves similar: If the function code is not reflected in the response, it's considered an "Unknown answer".
Nevertheless I encourage you to adding a method for hex dumping the response in every module action.

Reply:
Resolved with addition of `HEXDUMP` option.